src/clients/python/client.py

```diff
@@ -0,0 +1,179 @@
+import ctypes
+import threading
+import uuid
+
+from typing import List
+from queue import Queue
+import _tb_client
+
+ffi = _tb_client.ffi
+lib = _tb_client.lib
+
+def generate_enums():
+    enum_prefixes = ["TB_ACCOUNT", "TB_PACKET", "TB_OPERATION", "TB_CREATE_ACCOUNT", "TB_CREATE_TRANSFER", "TB_TRANSFER"]
+    enums = {}
+    enum_values = dir(lib)
+    for prefix in enum_prefixes:
+        if prefix not in enums:
+            enums[prefix] = {}
+        for val in enum_values:
+            if val.startswith(prefix):
+                enums[prefix][getattr(lib, val)] = val
+    return enums
+
+ENUMS = generate_enums()
+
+@ffi.def_extern()
+def on_completion_fn(context, client, packet, result_ptr, result_len):
+    """
+    Simple statically registered extern "Python" fn. This gets
+    called for any callbacks, looks up the respective client from our
+    global mapping, and forwards on the callback.
+
+    NB: This runs in the Zig's client thread.
+    """
+    client = Client.completion_mapping[client]
+    client._on_completion_fn(context, client, packet, result_ptr, result_len)
+
+class TigerBeetleInitException(Exception):
+    pass
+
+class Client:
+    completion_mapping = {}
+
+    def __init__(self, cluster_id: int, replica_addresses: List[str]):
+        self.tb_client = ffi.new("tb_client_t *")
+
+        self.out_packets = ffi.new("tb_packet_list_t *")
+        self.out_packets.head = ffi.new("tb_packet_t *")
+        self.out_packets.tail = ffi.new("tb_packet_t *")
+
+        status = _tb_client.lib.tb_client_init(
+                self.tb_client,
+                self.out_packets,
+                cluster_id,
+                b"3001",
+                4,
+                4096,
```

Comment:
You're initializing the `tb_client` with 4096 concurrent packets, but you are dynamically creating new packets on `_request` by doing `packet = ffi.new("tb_packet_t *")`.

You can either pass `0` to this parameter (no pre-allocated packets) and keep allocating them dynamically on python side -- or -- you can do it "TB way" and let the `tb_client` statically allocate all packets (returned on `out_packets` variable)  and then thread-safely acquire a packet from `packet_list` during `_request`, return it on `_on_completion_fn`.